daemon/logger/logbuffer/ringbuff.go

```diff
@@ -18,26 +23,33 @@ type RingBuffer struct {
 	cap    int
 	closed bool
 	q      *queue
+
+	maxBytes     int64
+	currentBytes int64
 }
 
 // New creates new RingBuffer.
-func New(cap int) *RingBuffer {
+func New(cap int, maxBytes int64) *RingBuffer {
 	if cap <= 0 {
 		cap = defaultSize
 	}
+	if maxBytes < 0 {
+		maxBytes = defaultMaxBytes
+	}
 
 	rb := &RingBuffer{
-		cap:    cap,
-		closed: false,
-		q:      newQueue(),
+		cap:      cap,
+		closed:   false,
+		q:        newQueue(),
+		maxBytes: maxBytes,
 	}
 	rb.wait = sync.NewCond(&rb.mu)
 	return rb
 }
 
 // Push pushes value into buffer and return whether it covers the oldest data
 // or not.
-func (rb *RingBuffer) Push(val interface{}) (bool, error) {
+func (rb *RingBuffer) Push(val *logger.LogMessage) (bool, error) {
```

Comment:
useless cover return-value here

Reply:
I agree with you, and the mentioned above has been updated. :)